### PR TITLE
Bug 1856438 - Reorg Search settings to match the new Cross Platform Suggest Design

### PR DESCRIPTION
**Pull request by tiftran**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1856438

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
Comment (linabutler) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt:41]: Let's get a follow-up bug on file to link to the final SUMO page, once https://bugzilla.mozilla.org/show_bug.cgi?id=1855091 is ready.

File: fenix/app/src/main/res/values/strings.xml
Comment (linabutler) [inline fenix/app/src/main/res/values/strings.xml:381]:
```suggestion
    <!-- Preference for settings related to Search engines -->
```

File: fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
Comment (linabutler) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt:129]: Nit: Stray blank line; ktlint should catch this! 😊

```suggestion
```

File: fenix/app/src/main/res/values/strings.xml
Comment (linabutler) [inline fenix/app/src/main/res/values/strings.xml:392]:
```suggestion
    <string name="preference_search_learn_about_fx_suggest">Learn more about Firefox Suggest</string>
```

Those spaces like to sneak in! 👀

File: fenix/app/src/main/res/values/strings.xml
Comment (linabutler) [inline fenix/app/src/main/res/values/strings.xml:388]: A question and a suggestion! 😊

* It looks like the only place that `preferences_search_address_bar` is used now is the old `search_preferences.xml`. Is it preferred to make a new string, instead of changing the value?
* Let's clarify in this comment (and the one for the Firefox Suggest section below) that this is a preference category, not the preference itself.

File: fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
Comment (linabutler) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt:1]: Could you please add a bit of code to only show your new `pref_key_learn_about_fx_suggest` menu item if the `enableFxSuggest` flag is set?

`onCreatePreferences` does this for `pref_key_show_sponsored_suggestions` and `pref_key_show_nonsponsored_suggestions` now; you can add your code just below those!

File: fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
Comment (linabutler) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt:41]: We might also want to use `getSumoURLForTopic` or `getGenericSumoURLForTopic` here, instead of hard-coding the `en-US` URL (even though it is US-only to start with).

Let's see what the mobile folks think!

**Review by linabutler — COMMENTED** [review]
Thanks! It's so exciting to see this come together! 🚀 Just a quick pass to catch a few nits...

File: fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
Comment (tiftran) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt]: oooh i missed `getGenericSumoURLForTopic` when i was first digging around. Makes sense to me, but still curious what mobile folks think.

File: fenix/app/src/main/res/values/strings.xml
Comment (tiftran) [reply fenix/app/src/main/res/values/strings.xml:388]: oh is `search_preferences.xml` no longer used? My first assumption that since there was usage for it elsewhere, i didn't want to disturb it.

**Comment by tiftran** [comment]
ohhh eeek, I can't request anyone to review so tagging here instead @MatthewTighe.
Also, silly question, is it convention to squash commits from feedback into the original commit? or is it okay to squash after the PR is good to merge?

File: fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
Comment (linabutler) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt:50]:
```suggestion
        requirePreference<Preference>(R.string.pref_key_learn_about_fx_suggest).apply {
```

I think we'll crash with a bad cast exception if the type passed to `requirePreference` doesn't match the type of the element in the XML.

Comment (tiftran) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt:50]: ooops

File: fenix/app/src/test/java/org/mozilla/fenix/settings/search/SearchEngineFragmentTest.kt
Comment (linabutler) [inline fenix/app/src/test/java/org/mozilla/fenix/settings/search/SearchEngineFragmentTest.kt:102]:
```suggestion
                fragment.findPreference<Preference>(testContext.getString(R.string.pref_key_learn_about_fx_suggest))
```

**Review by linabutler — APPROVED** [review]
This looks great to me, but let's wait for @MatthewTighe to review! ☺️

Here's a screenshot of the new Search settings:

settings

File: fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
Comment (MatthewTighe) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt]: I'm unaware of how SUMO handles translations, but it looks like replacing the locale in the URL above will link to translated pages, for example here the link above with the locale switched to Spanish: https://support.mozilla.org/es/kb/search-suggestions-firefox. 

Not sure if there is some auto-translate magic happening or if there are already existing translated pages for that article. I would say that using `getGenericSumoURLForTopic` looks the most future proof, but I am not sure what happens if we generate a link for a locale that does not have a translation available.

File: fenix/app/src/main/res/values/strings.xml
Comment (MatthewTighe) [reply fenix/app/src/main/res/values/strings.xml:388]: `search_preferences.xml` isn't currently used, though we could fall back to it through a Nimbus flag if we needed so it isn't dead code quite yet. 

Generally speaking we prefer to create new strings rather than update existing ones. If a string is no longer used, we mark it with the xml tag `moz:RemovedIn="<version>". This avoids errors with reversions, depending on where a patch is in regards to the trains.

**Review by MatthewTighe — APPROVED** [review]
LGTM! Thanks for handling this piece 🙂

File: fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt:112]: nit let's not remove this new line

File: fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
Comment (linabutler) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt:41]:
```suggestion

```

Let's fix up the trailing whitespace 👀

File: fenix/app/src/main/res/xml/search_settings_preferences.xml
Comment (gabrielluong) [inline fenix/app/src/main/res/xml/search_settings_preferences.xml:22]: nit: Could we keep this new line to help maintain that visual separation of our `<PreferenceCategory>`? This will mean also add a new line for for your new additions as well.

File: fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/SupportUtils.kt:41]: nit: not sure if we should keep this?

File: fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
Comment (tiftran) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt:112]: eek will do, sorry, lemme fix up the spacing and extra lines. Tyty for taking the time to review 🙂

File: fenix/app/src/main/res/values/strings.xml
Comment (gabrielluong) [inline fenix/app/src/main/res/values/strings.xml:388]:
```suggestion
    <string name="preferences_settings_address_bar" moz:removedIn="120" tools:ignore="UnusedResources">Address bar preferences</string>
```
I goofed here and read the calendar wrong

Comment (tiftran) [reply fenix/app/src/main/res/values/strings.xml]: realized i also goofed since it's the line above that's no longer in use

File: fenix/app/src/main/res/values/strings.xml
Comment (gabrielluong) [inline fenix/app/src/main/res/values/strings.xml:386]:
```suggestion
    <string name="preferences_search_address_bar" moz:removedIn="120" tools:ignore="UnusedResources">Address bar</string>
```

**Comment by gabrielluong** [comment]
No additional review needed from me!

File: fenix/app/src/main/res/values/strings.xml
Comment (tiftran) [reply fenix/app/src/main/res/values/strings.xml:386]: 🤦‍♀️ ugh, sorry

**Pull request merged**
